fairlearn/postprocessing/_threshold_optimizer.py

```diff
@@ -109,8 +109,10 @@ class ThresholdOptimizer(BaseEstimator, MetaEstimatorMixin):
 
     Parameters
     ----------
-    estimator : estimator object implementing 'predict' and possibly 'fit'
-        An estimator whose output is postprocessed.
+    estimator : object
+        An estimator whose output is postprocessed. The estimator should be
+        implementing a :code:`predict` method. If :code:`prefit=False`, it
+        also needs to implement a :code:`fit` method.
```

Comment:
It's probably easier to say "a scikit-learn compatible estimator". Since there's more to it than fit and predict for it to be something which can then be used in a `Pipeline` or a `GridSearchCV` for instance.

Reply:
I'm fine either way, but I'm not sure whether we require an estimator. With `prefit=True`, I think it just needs to be an object that supports predict.

Reply:
I was thinking along the same lines as @MiroDudik here. I think in scikit-learn terms we require
- a predictor if `prefit=True`
- an estimator if `prefit=False`

https://scikit-learn.org/stable/developers/develop.html?highlight=predictor#different-objects

Reply:
I would say  we just say sklearn compatible, and only officially support that.

Reply:
Replaced as you suggested! This should be future-proof 🙂